src/zkevm_specs/evm/execution/chainid.py

```diff
@@ -0,0 +1,21 @@
+from zkevm_specs.util.param import N_BYTES_WORD
+from ..instruction import Instruction, Transition
+from ..table import BlockContextFieldTag
+from ..opcode import Opcode
+
+
+def chainid(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.CHAINID)
```

Comment:
Whether the opcode is responsible by this `ExecutionState` is implicitly checked in `step_state_transition_in_same_context`, so this could be removed.

(btw I was extracting the logic from `step_state_transition_in_same_context` previously but haven't done yet, because its name seems not straightforward to include such check (my apology), also I think the constant gas cost part should be explicitly set.)

Reply:
> Whether the opcode is responsible by this `ExecutionState` is implicitly checked in `step_state_transition_in_same_context`, so this could be removed.

That was my initial thought as well, but then I saw that the other opcodes also have this constraint in the python spec, so I thought I was mistaken.  Maybe we should open an issue to remove that constraint from the rest of the opcodes?

Reply:
My thought was the same as Edu's. That's why I did not proceed with extra modifications.

But happy to help on the refactor!